src/main/java/org/gwtproject/activity/widget/shared/WidgetActivityManager.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2010 Google Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.gwtproject.activity.widget.shared;
+
+import org.gwtproject.activity.shared.ActivityDisplay;
+import org.gwtproject.activity.shared.ActivityManager;
+import org.gwtproject.event.shared.EventBus;
+import org.gwtproject.user.client.ui.AcceptsOneWidget;
+import org.gwtproject.user.client.ui.IsWidget;
+
+/**
+ * Drop-in replacement for {@link com.google.gwt.activity.shared.ActivityManager}.
+ */
+public class WidgetActivityManager extends ActivityManager<IsWidget> {
+
+  private AcceptsOneWidget widgetDisplay;
+
+  private ActivityDisplay<IsWidget> wrapperDisplay;
+
+  public WidgetActivityManager(WidgetActivityMapper mapper, EventBus eventBus) {
+    super(mapper, eventBus);
+  }
+
+  public void setDisplay(AcceptsOneWidget display) {
```

Comment:
Should this extend or wrap?

* by extending, this allows any `ActivityDisplay<IsWidget>`, but breaks `setDisplay(null)` that now requires a explicit cast to choose an overload
* by wrapping, it cannot be used interchangeably with an `ActivityManager<IsWidget>` (similar to the `CompatEventBus` in gwt-event-compat), but ensures backwards compatibility

Reply:
Like you pointed out, either one is bad. Javadoc could alleviate issue with required cast of `null`s, but it's far from optimal. What is your suggestion?

Reply:
I would wrap, because I don't see a need for casting to `ActivityManager<IsWidget>`. And I would have a constructor that takes the `ActivityManager<IsWidget>` to be wrapped, and another constructor that would instantiate one, for ease of use. If one needs both an `ActivityManager<IsWidget>` and a `WidgetActivityManager`, they could then create the former and wrap it in the latter (similar to `CompatEventBus`)

Reply:
Wrapped.

I exposed all public methods from wrapped `ActivityManager<IsWidget>` together with wrapped object itself (via (`getDelegateManager`) and explained relationship to `ActivityManager<IsWidget>` in docs.